src/subsplit_dag.cpp

```diff
@@ -15,11 +15,13 @@ SubsplitDAG::SubsplitDAG(size_t taxon_count,
   Assert(topology_counter.size() > 0, "Empty topology counter given to SubsplitDAG.");
   Assert(topology_counter.begin()->first->LeafCount() == taxon_count,
          "Taxon count mismatch in SubsplitDAG constructor.");
-  ProcessTopologyCounter(topology_counter);
-  BuildNodes();
-  BuildEdges();
-  ExpandRootsplitsInIndexer();
-  AddFakeSubsplitsToGPCSPIndexerAndParentToRange();
+  BitsetSizeMap gpcsp_indexer;
+  SizeBitsetMap index_to_child;
+  ProcessTopologyCounter(topology_counter, gpcsp_indexer, index_to_child);
```

Comment:
Passing an "out" argument should be done by return (https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-out)

Can you refactor this so that the indexer and `index_to_child` are passed out? I think we can use an auto structured binding to get them.